manifests/config.pp

```diff
@@ -89,7 +89,7 @@
     ensure => 'directory',
     owner  => 'root',
     group  => $auditd::log_group,
-    mode   => 'o-rwx'
+    mode   => $log_file_mode,
```

Comment:
This should have been line 48, right?

Reply:
48 controls the /etc/audit/autid.rules.  This line controls the directory /var/log/audit.  The issue I'm trying to correct is I have a role account in the $auditd::log_group that is unable to traverse /var/log/audit and so cannot access any of the logs even though all the logs are owned by that group.

I did not run into any problems with the /etc/audit/audit.rules, I had just noticed it was similarly controlled by removing other permissions and not setting group permissions.  To keep this pull simple I removed that change.

Reply:
@zschulte Sorry, I think I may have been a bit confusing.

I think that `log_file_mode` may just be badly named. What I'm getting at is that the auditd configuration files and the auditd logs should not necessarily have the same permissions so we need a second variable.

Something like `config_file_mode` vs `log_file_mode` might do the trick but I'll have to check for backward compatibility issues.

Reply:
I understand.  That is why I removed the change to the config file and left the change to the log directory's permissions.

Currently, this pull will only change the log directory /var/log/audit to be controlled by the variable $log_file_mode.

Reply:
@trevor-vaughan it has been a while and we just rebased this to see if we can get this pushed through. Are the changes @zschulte made good, or is there something else that is needed here?

Thanks!

Reply:
@trevor-vaughan I just pushed an update (explained in new comment). Can we look into getting this pushed along again? Sorry for the delay.